src/driver/mod.rs

```diff
@@ -68,15 +70,19 @@ cfg_if::cfg_if! {
 /// ```
 pub trait Poller {
     /// Attach an fd to the driver.
+    ///
+    /// ## Platform specific
+    /// * IOCP: it will be attached to the IOCP completion port.
+    /// * io-uring: it will do nothing and return `Ok(())`
     fn attach(&self, fd: RawFd) -> io::Result<()>;
 
     /// Push an operation with user-defined data.
     /// The data could be retrived from [`Entry`] when polling.
     ///
     /// # Safety
     ///
-    /// `op` should be alive until [`Poller::poll`] returns its result.
-    unsafe fn push(&self, op: &mut (impl OpCode + 'static), user_data: usize) -> io::Result<()>;
+    /// - `op` should be alive until [`Poller::poll`] returns its result.
+    unsafe fn push<T: OpCode + 'static>(&self, op: &mut T, user_data: Key<T>) -> io::Result<()>;
```

Comment:
`Poller` is a low-level interface. Should it be aware of the relationship between op and user_data?

Reply:
Good point. I'll change it back then.